core/server/worker/src/main/java/alluxio/worker/grpc/AbstractReadHandler.java

```diff
@@ -439,8 +485,12 @@ private void replyEof() {
       mSerializingExecutor.execute(() -> {
         try {
           Preconditions.checkState(!mContext.isDoneUnsafe());
-          mContext.setDoneUnsafe(true);
-          mResponse.onCompleted();
+          if (!mContext.isDoneUnsafe()) {
+            mContext.setDoneUnsafe(true);
+            mResponse.onCompleted();
+          } else {
+            LOG.warn("Tried to replyEof when stream was already finished. context: {}", mContext);
```

Comment:
When will this happen? Will it be too chatty on the log?

Reply:
From all of my testing, I never saw any of these logs appear. It's possible there's cases I haven't triggered yet. But I have about 500MB of worker logs from my testing and I don't see a single one where we tried to reply twice

Reply:
I can move this to debug logging? or remove it all together. In the 500M of logs I have, I only seen the `replyError` one